agent/src/main/scala/za/co/absa/atum/agent/AtumContext.scala

```diff
@@ -54,6 +64,17 @@ class AtumContext private[agent] (
     }
   }
 
+  /**
+   * Creates a checkpoint in the AtumContext.
+   *
+   * This method is used to mark a specific point in the data processing pipeline where measurements of data completeness are taken.
```

Comment:
"of data completeness" - not sure if we want to be this specific here 🤔

Reply:
I think it's reasonable. But if we find it too wordy/detailed/specific we can alter the doc string in the future. Let's keep it for now.